opencti-platform/opencti-graphql/src/migrations/1718263084981-update-data-sharing-ingestion-capabilities.js

```diff
@@ -0,0 +1,82 @@
+import { logApp } from '../config/conf';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+import { elList, elLoadById, elReplace } from '../database/engine';
+import { roleCapabilities } from '../domain/user';
+import { addCapability } from '../domain/grant';
+import { createRelation } from '../database/middleware';
+import { ENTITY_TYPE_ROLE } from '../schema/internalObject';
+import { READ_INDEX_INTERNAL_OBJECTS } from '../database/utils';
+
+const message = '[MIGRATION] Split data sharing & ingestion into 2 capabilities';
+
+export const up = async (next) => {
+  logApp.info(`${message} > started`);
+  const context = executionContext('migration');
+
+  // ------ Create Access ingestion
+  const accessIngestionCapability = await addCapability(context, SYSTEM_USER, {
+    name: 'INGESTION',
+    attribute_order: 2600,
+    description: 'Access ingestion'
+  });
+  // ------ Create Manage ingestion
+  const manageIngestionCapability = await addCapability(context, SYSTEM_USER, {
+    name: 'INGESTION_SETINGESTIONS',
+    description: 'Manage ingestion',
+    attribute_order: 2610
+  });
+
+  // ------ Update roles
+  const callback = async (roles) => {
+    for (let i = 0; i < roles.length; i += 1) {
+      const roleId = roles[i].id;
+      const capabilities = await roleCapabilities(context, SYSTEM_USER, roleId);
+      // Select 'Access ingestion' if 'Access Data sharing & ingestion' is selected
+      const hasAccessDataSharingCapability = capabilities.some((capability) => capability.name === 'TAXIIAPI');
+      if (hasAccessDataSharingCapability) {
+        const input = { fromId: roleId, toId: accessIngestionCapability.id, relationship_type: 'has-capability' };
+        await createRelation(context, SYSTEM_USER, input);
+      }
+      // Select 'Manage ingestion' if 'Manage Data sharing & ingestion' is selected
+      const hasManageDataSharingCapability = capabilities.some((capability) => capability.name === 'TAXIIAPI_SETCOLLECTIONS');
+      if (hasManageDataSharingCapability) {
+        const input = { fromId: roleId, toId: manageIngestionCapability.id, relationship_type: 'has-capability' };
+        await createRelation(context, SYSTEM_USER, input);
+      }
+    }
+  };
+  const opts = { types: [ENTITY_TYPE_ROLE], callback };
+  await elList(context, SYSTEM_USER, READ_INDEX_INTERNAL_OBJECTS, opts);
+
+  // ------ Update Attribute_order and name of Manage CSV mappers
+  const CSVCapability = await elLoadById(context, SYSTEM_USER, 'capability--5407e8a4-0ff9-5253-89b1-c01a92ad9453');
+  const CSVCapabilityPatch = { name: 'INGESTION_SETCSVMAPPERS', attribute_order: 2620 };
+  await elReplace(CSVCapability._index, CSVCapability.internal_id, { doc: CSVCapabilityPatch });
```

Comment:
when we rename a capability, shouldn't we change also its standard_id ? Locally, after executing the migration, the capability name has been updated but standard id has not changed.

Reply:
we need to update the standard_id. there is an example here: 1714034457885-fix-knowledge-orga-restrict-capability-name.js

Reply:
Done